### PR TITLE
Enable type-based linter rules

### LINKED ISSUE #1744: Enable lint rule: @typescript-eslint/strict-boolean-expressions
We have [a bug](https://github.com/veloce/lichobile/issues/1727) probably caused by using non-booleans in conditionals. Let's enable [@typescript-eslint/strict-boolean-expressions](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md), with options as necessary, to avoid these in the future.

### PR DESCRIPTION
**Pull request by jas14**
And upgrade NPM. Closes #1744.

### CODE REVIEW

**Comment by jas14** [comment]
https://github.com/npm/cli/issues/2610

File: src/chessground/drag.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ export function end(ctrl: Chessground, e: TouchEvent) {
   // board editor mode: delete any piece dropped off the board
   else if (cur.started && draggable.deleteOnDropOff) {
     state.pieces.delete(cur.orig)
-    setTimeout(state.events.change || util.noop, 0)
+    setTimeout(() => (state.events.change || util.noop)(), 0)
```
Comment (veloce) [inline src/chessground/drag.ts:181]: why this change?

Comment (jas14) [reply src/chessground/drag.ts:181]: https://eslint.org/docs/rules/no-implied-eval

Comment (veloce) [reply src/chessground/drag.ts:181]: but I don't get it, as here this is not a string argument that is passed to `setTimeout`, it is a function. I'm not a fan of this change really.

Comment (jas14) [reply src/chessground/drag.ts:181]: I think you're right, it's a bug in the rule... the type of the expression before this change is `() => void`, and you can "fix" the error by assigning the expression to a const first
```typescript
const fn = state.events.change || util.noop
setTimeout(fn, 0)
```

Comment (veloce) [reply src/chessground/drag.ts:181]: then I much prefer to have this instead of the arrow function wrapper

Comment (jas14) [reply src/chessground/drag.ts:181]: https://github.com/typescript-eslint/typescript-eslint/issues/3621

Comment (jas14) [reply src/chessground/drag.ts]: Just backed out this change, we can re-enable the rule later once the bug is fixed
